### PR TITLE
Use HDBSCAN for clustering

### PR DESCRIPTION
**Pull request by marcelklehr**
Thanks to @MB-Finski

### CODE REVIEW

File: lib/Service/FaceClusterAnalyzer.php [hunk 1]
```diff
@@ -484,71 +54,71 @@ public function calculateClusters(string $userId): void {
...
 		$detections = $this->faceDetections->findByUserId($userId);
 
-		if (count($detections) < max(self::MIN_SAMPLE_SIZE, self::MIN_CLUSTER_SIZE)) {
+		$detections = array_values(array_filter($detections, fn ($detection) =>
+			$detection->getHeight() > self::MIN_DETECTION_SIZE && $detection->getWidth() > self::MIN_DETECTION_SIZE
+		));
+
+		$unclusteredDetections = $this->assignToExistingClusters($userId, $detections);
+
+		if (count($unclusteredDetections) < max(self::MIN_SAMPLE_SIZE, self::MIN_CLUSTER_SIZE)) {
```
Comment (MB-Finski) [inline lib/Service/FaceClusterAnalyzer.php:54]: Before soft clustering is implemented it migh make sense to use a much higher limit here for  running HDBSCAN on the "left overs". This is due to the fact that HDBSCAN learns the correct cluster limits from the data. If it's fed garbage, it'll still try and find stable clusters from that garbage.

Comment (MB-Finski) [reply lib/Service/FaceClusterAnalyzer.php:54]: Even feeding in some thousands of detections from the already clustered detections to assist HDBSCAN make sense of the data could be a good interim solution.

**Comment by kc9jud** [comment]
Referenced in discussion #630. (Apparently GitHub doesn't automatically trackback to discussions?)

File: lib/Service/FaceClusterAnalyzer.php (hunk 1 above)
Comment (marcelklehr) [reply lib/Service/FaceClusterAnalyzer.php]: I'm sampling 42 random detections from each existing cluster now

**Pull request merged**